src/main/java/igrad/model/requirement/Credits.java

```diff
@@ -5,24 +5,26 @@
 
 /**
  * Represents a Requirement's required credit units in the course book.
- * Guarantees: immutable; is valid as declared in {@link #isValidCredits(String)}
+ * Guarantees: immutable, non-null and is valid as declared by {@link #isValidCredits(String)}
  */
 public class Credits {
 
-    public static final String MESSAGE_CONSTRAINTS = "Modular credits should contain one or more digits!";
+    public static final String MESSAGE_CONSTRAINTS =
+        "Credits needed to satisfy requirement should be more than 0.";
 
-    public static final String VALIDATION_REGEX = "\\d+";
+    public static final String VALIDATION_REGEX = "^[0-9]\\d*$"; // allows any numbers more than or equals zero
```

Comment:
hmm why equals to zero though, no modules can be  added under it then when i start to work on assign haha